cargo-libafl/cargo-libafl-runtime/runtime.rs

```diff
@@ -153,51 +162,75 @@ pub fn main() {
+
     let workdir = env::current_dir().unwrap();
 
     let opt = Opt::parse();
 
     let cores = opt.cores;
     let broker_port = opt.broker_port.unwrap_or_else(|| {
         let port = portpicker::pick_unused_port().expect("No ports free");
-        println!("Picking the free port {}", port);
+        log::info!("Picking the free port {}", port);
         port
     });
     let remote_broker_addr = opt.remote_broker_addr;
-    let input_dirs = opt.input;
-    let output_dir = opt.output;
+    let mut input_dirs = opt.input.clone();
+    let corpus_dir = opt.corpus;
+    let crashes_dir = opt.crashes;
     let token_files = opt.tokens;
     let timeout_ms = opt.timeout;
-    // let cmplog_enabled = matches.is_present("cmplog");
 
-    if fs::create_dir(&output_dir).is_err() {
-        println!("Out dir at {:?} already exists.", &output_dir);
-        if !output_dir.is_dir() {
-            eprintln!("Out dir at {:?} is not a valid directory!", &output_dir);
-            return;
+    for dir in &[&corpus_dir, &crashes_dir] {
+        if fs::create_dir(dir).is_err() {
+            log::warn!("Out dir at {:?} already exists.", dir);
+            if !dir.is_dir() {
+                log::error!("Required directory at {:?} is not a valid directory!", dir);
+                return;
+            }
         }
     }
-    let crashes_dir = output_dir.join("crashes");
-    let corpus_dir = output_dir.join("corpus");
-
-    println!("Workdir: {:?}", workdir.to_string_lossy().to_string());
+    log::info!(
+        "Workdir: {:?}, Corpus: {:?}, Crashes: {:?}",
+        workdir.to_string_lossy().to_string(),
+        corpus_dir,
+        crashes_dir
+    );
 
     let shmem_provider = StdShMemProvider::new().expect("Failed to init shared memory");
 
-    let monitor = TuiMonitor::new(format!("cargo-libafl v{}", VERSION), !opt.disable_unicode);
+    #[cfg(feature = "tui")]
+    let monitor = {
+        use libafl::monitors::tui::{ui::TuiUI, TuiMonitor};
+        let ui = TuiUI::with_version(
+            String::from("cargo-libafl"),
+            VERSION.to_string(),
+            !opt.disable_unicode,
+        );
+        TuiMonitor::new(ui)
+    };
+    #[cfg(not(feature = "tui"))]
+    let monitor = libafl::monitors::MultiMonitor::new(|s| log::info!("{}", s));
 
     let mut run_client = |state: Option<StdState<_, _, _, _>>, mut mgr, _core_id| {
-        // Create an observation channel using the coverage map
-        let edges = unsafe { &mut COUNTERS_MAPS };
-        let edges_observer =
-            HitcountsIterableMapObserver::new(MultiMapObserver::new("edges", edges));
+        log::debug!("running fuzzing client");
+        // first create an observation channel using the coverage map
+
+        #[cfg(feature = "sancov_8bit")]
+        let edges_observer = {
+            let edges = unsafe { &mut COUNTERS_MAPS };
+            // TODO: is the call to edges.clone here the reason for breaking sancov_8bit?
```

Comment:
Yes
After edges.clone(), `OwnedMutSlice::RefRaw` becomes `OwnedMutSliced::Owned`.
then it does not have the reference to the original edge counter

Reply:
I see. It works, when I construct a new `OwnedMutSlice` from the raw parts of the previous one. But I still don't understand how the `OwnedMutSlice` is supposed to be used in this case. Rather move the `OwnedMutSlice` out of the `COUNTER_MAPS` and into the `edges` parameter of the `MultiMapObserver`? 

Why doesn't the `MultiMapObserver` to receive a reference to the `COUNTER_MAPS`? Is it important that the number/size of the maps don't change while fuzzing?